pkg/replication/replication.go

```diff
@@ -742,9 +748,109 @@ type ResyncTarget struct {
 	FailedSize int64 `json:"failedReplicationSize,omitempty"`
 	// Total number of failed operations
 	FailedCount int64 `json:"failedReplicationCount,omitempty"`
-	// Total number of failed operations
+	// Total number of completed operations
 	ReplicatedCount int64 `json:"replicationCount,omitempty"`
 	// Last bucket/object replicated.
 	Bucket string `json:"bucket,omitempty"`
 	Object string `json:"object,omitempty"`
 }
+
+// XferStats holds transfer rate info for uploads/sec
+type XferStats struct {
+	AvgRate  float64 `json:"avgRate"`
+	PeakRate float64 `json:"peakRate"`
+	CurrRate float64 `json:"currRate"`
+}
+
+// InQueueStats holds stats for objects in replication queue
+type InQueueStats struct {
+	Count int32 `json:"count"`
+	Bytes int64 `json:"bytes"`
+}
+
+// ReplQNodeStats holds stats for a node in replication queue
+type ReplQNodeStats struct {
+	NodeName      string `json:"nodeName"`
+	Uptime        int64  `json:"uptime"`
+	ActiveWorkers int32  `json:"activeWorkers"`
+
+	// transfer stats for objects larger/smaller than 128 MiB
+	Large XferStats `json:"Large"`
+	Small XferStats `json:"Small"`
+
+	// queue stats for objects currently in replciation queue
+	QTotals InQueueStats `json:"queueTotals"`
+	QLarge  InQueueStats `json:"queuedLarge"`
+	QSmall  InQueueStats `json:"queuedSmall"`
+}
+
+// ReplQueueStats holds stats for replication queue across nodes
+type ReplQueueStats struct {
+	Nodes []ReplQNodeStats `json:"nodes"`
+}
+
+// Workers returns number of workers across all nodes
+func (q ReplQueueStats) Workers() int64 {
+	var workers int64
+	for _, node := range q.Nodes {
+		workers += int64(node.ActiveWorkers)
+	}
+	return workers
+}
+
+// ReplQStats holds stats for objects in replication queue
+type ReplQStats struct {
+	Uptime  int64 `json:"uptime"`
+	Workers int64 `json:"workers"`
+	// transfer stats for objects larger/smaller than 128 MiB
+	Large XferStats `json:"large"`
+	Small XferStats `json:"small"`
+	// queue stats for objects currently in replication queue
+	QTotals InQueueStats `json:"queueTotals"`
+	QLarge  InQueueStats `json:"queuedLarge"`
+	QSmall  InQueueStats `json:"queuedSmall"`
```

Comment:
My guess is that this can change in the future. To avoid the struggle of compatibility issues, it looks like it is easier to have on field:
map[string]InQueueStats (or int instead of string)